### ISSUE TITLE
Some Parameter types require ``default=`` (e.g. Dict), but most don't

### ISSUE THREAD

**Issue by ceball** [holoviz/param#91]
``` python
>>> import param
>>> param.List([1,2,3])
<param.List object at 0x0000000004D62BF8>
>>> param.String('1 2 3')
<param.parameterized.String object at 0x00000000029ED930>
>>> param.Dict({1:2})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() takes exactly 1 argument (2 given)
>>> param.Dynamic(10)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() takes exactly 1 argument (2 given)
>>> param.Array(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() takes exactly 1 argument (2 given)
```

**Comment by ceball** [comment]
It's either very confusing or it's annoying, depending on how long you've been using param...

**Comment by ceball** [comment]
Closed in favor of #393
